spec/validators/email_mx_validator_spec.rb

```diff
@@ -41,6 +41,16 @@
       expect(user.errors).to_not have_received(:add)
     end
 
+    it 'adds an error if the domain resolves as blank' do
+      double = instance_double(TagManager)
+      allow(TagManager).to receive(:instance).and_return(double)
+      allow(double).to receive(:normalize_domain).with('example.com').and_raise(Addressable::URI::InvalidURIError)
```

Comment:
I'm not completely sure what this example is about. The description is “if the domain resolves as blank” but the test appears to be testing something else.

Furthermore, instead of stubbing `TagManager.instance#normalize_domain` to raise `Addressable::URI::InvalidURIError` for a perfectly valid domain, I think it would make more sense to come up with an invalid domain as input in the first place.

Reply:
> I'm not completely sure what this example is about. The description is “if the domain resolves as blank” but the test appears to be testing something else.

Good point ... I named the spec first because my intention was to exercise the very first code path in the validator which does the `if domain.blank?` check. This was previously not being run. Once I started looking for how to do that, I saw that the error catching line was also not being hit and found this spec to hit them both at once.

I just pushed a commit which changes things to have two specs here -- one which has a blank domain on the email argument sent in, and one which exercises the error catching path. They are hopefully better named now as well.